cluster/cluster.go

```diff
@@ -598,6 +601,30 @@ func parseIngressExtraVolumeMounts(ingressMap map[string]interface{}, rkeConfig
 	return nil
 }
 
+func parseIngressDefaults(ingressMap map[string]interface{}, rkeConfig *v3.RancherKubernetesEngineConfig) error {
+	// Setting up default behaviour so as to not catch out
+	// existing users who use new version of RKE
+	if _, ok := ingressMap["hostNetwork"]; !ok {
+		rkeConfig.Ingress.HostNetwork = true
+	}
+	if _, ok := ingressMap["http_port"]; !ok {
+		rkeConfig.Ingress.HTTPPort = 80
+	}
+	if _, ok := ingressMap["https_port"]; !ok {
+		rkeConfig.Ingress.HTTPSPort = 443
+	}
+
+	if !(rkeConfig.Ingress.HTTPSPort >= 0 && rkeConfig.Ingress.HTTPSPort <= 65535) {
+		return fmt.Errorf("[parseIgressDefaults] https port is invalid. Needs to be within 0 to 65535")
+	}
+
+	if !(rkeConfig.Ingress.HTTPPort >= 0 && rkeConfig.Ingress.HTTPPort <= 65535) {
+		return fmt.Errorf("[parseIgressDefaults] http port is invalid. Needs to be within 0 to 65535")
+	}
+
```

Comment:
HTTP and HTTPS port can't be the same, need to add a validation.

Reply:
Added the extra validation.